### ISSUE TITLE
Usage of .NET API documentation

### ISSUE THREAD

**Issue by MazeXP** [lunet-io/lunet#4]
Hi Alexande,

I wanted to ask you if you have any sample on how to use the .Net API documentation generation.

I was able to have it generate the *.api.json file, but that is actually all about it, not a single page.

That is my added portion to the config.scriban:
```cs
with api.dotnet
	projects = [
		"../Backend/Remora.Discord.API.Abstractions/Remora.Discord.API.Abstractions.csproj"
	]

	properties = {
		TargetFramework: "net6.0"
	}
end
```

I get following warning:
```
warn: lunet[4] Multiple api dotnet output generated for `Remora.Discord.API.Abstractions`. Consider setting TargetFramework in the config to use only one output. Using by default C:\Users\MazeXD\RiderProjects\Remora.Discord\Backend\Remora.Discord.API.Abstractions\obj\Release\net6.0\Remora.Discord.API.Abstractions.api.json
```

I verified the output of the dotnet build command.
**Command**
```batch
dotnet build -c Release Remora.Discord.API.Abstractions.csproj -p:Configuration=Release -p:CustomBeforeMicrosoftCommonProps="%userprofile%\.dotnet\tools\.store\lunet\0.4.2\lunet\0.4.2\tools\net5.0\any\shared\.lunet\modules\api\dotnet\Lunet.Api.DotNet.Extractor.props" -p:TargetFramework=net6.0
```
**Output**
```
Microsoft (R) Build Engine version 17.2.0+41abc5629 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
CSC : warning XDoc0001: "C:\Users\MazeXD\RiderProjects\Remora.Discord\Backend\Remora.Discord.API.Abstractions\obj\Relea
se\net6.0\Remora.Discord.API.Abstractions.api.json" [C:\Users\MazeXD\RiderProjects\Remora.Discord\Backend\Remora.Discor
d.API.Abstractions\Remora.Discord.API.Abstractions.csproj]
  Remora.Discord.API.Abstractions -> C:\Users\MazeXD\RiderProjects\Remora.Discord\Backend\Remora.Discord.API.Abstractio
  ns\bin\Release\net6.0\Remora.Discord.API.Abstractions.dll

Build succeeded.

CSC : warning XDoc0001: "C:\Users\MazeXD\RiderProjects\Remora.Discord\Backend\Remora.Discord.API.Abstractions\obj\Relea
se\net6.0\Remora.Discord.API.Abstractions.api.json" [C:\Users\MazeXD\RiderProjects\Remora.Discord\Backend\Remora.Discor
d.API.Abstractions\Remora.Discord.API.Abstractions.csproj]
    1 Warning(s)
    0 Error(s)

Time Elapsed 00:00:02.33
```

In addition it seems like the `.api.json` is not cached in the correct location as I would have expected that it should be cached under `[base]\.lunet\build\cache` but instead it is cache under `[base]\api\dotnet`.

If you need more information I am open to provide more.

**Comment by xoofx** [comment]
The API Documentation generation is unfortunately not finished yet, I stopped on my way and I haven't had the time/motivation to go back to it since then.

Many things are not working/not polished to use it afair. I pushed an example on a Markdig branch/commit [here](https://github.com/xoofx/markdig/commit/62907f9314faf8eb737175dd533000220ec48aff)

I think the API is by default generated to a fix path (e.g `api/readme.md`) that needs to be referenced from a menu in order to get access to it.

Other than that, it's generating lots of errors because of many generic parameters and certain constructions not correctly handled.
